### PR TITLE
Classifier fixes

### PR DESCRIPTION
**Pull request by PiotrCzapla**

### CODE REVIEW

**Comment by PiotrCzapla** [comment]
@NirantK @sebastianruder  , lets have a space to discuss the changes if needed.

File: fastai_contrib/utils.py [hunk 1]
```diff
@@ -3,20 +3,207 @@
...
+    processor = processors[dataset]
+
+    toks, lbls = {}, {}
+    toks[TRN], lbls[TRN] = processor(dir_path, lang, TRN)
+    toks[TST], lbls[TST] = processor(dir_path, lang, TST)
+
+    if dataset == IMDB:
+        # for IMDb, we need to split off a separate validation set
+        # note that we train and fine-tune ULMFiT on the full training set in the paper
+        # to do this, we can just keep the training set the same
```
Comment (PiotrCzapla) [inline fastai_contrib/utils.py]: @sebastianruder What do you mean by keeping the "training set the same"? You mean that we don't shuffle and we take first 90%  of it?

Comment (sebastianruder) [reply fastai_contrib/utils.py]: Just that we train on the full training set, so we don't split anything off. As we still need a validation set in the API, we can use examples from the training set, but the val loss will be meaningless.

**Comment by PiotrCzapla** [comment]
@sebastianruder @NirantK  I'm testing the whole pipeline for creating wikipedia data sets. One thing I'd like to change is to move the results of "postprocess_wikitext.py" to its own folders. 
Current approach creates `.unk`  files next to original files. Having that in a separate folder would let us train an LMs either on postprocessed text or on raw text. This might be useful as postprocessing is removing numbers and XNLI contains numbers. So LM  that understands what 2012 means might work better.  Let me know your thoughts.

**Comment by PiotrCzapla** [comment]
Btw. I've added another category `all` that holds everything that wikipedia has to offer. In our experiments on polish language modeling, we've found out that using the whole text improves perplexity, and if we limit our selves to 100m tokens per epoch we don't extend the training time. So I thought we might try that on other languages as well.

**Comment by sebastianruder** [comment]
@PiotrCzapla, `replace_numbers` actually does not replace numbers, but just splits long numbers, e.g. `300,000`/`300.000` into `300 @,@ 000`/`300 @.@ 000` as this is what is done by Merity et al. to process the WikiText datasets. Regular numbers such as `300` are left the same. I think we should stick with that. 
I agree with your point re training models on the entire Wikipedia, though.

**Comment by PiotrCzapla** [comment]
Directory structure after changes to the way we process wiki dumps.
```
data
├── en
│   └── wikitext-2
│       └── models
├── imdb
│   ├── aclImdb
│   │   ├── test
│   │   │   ├── all
│   │   │   ├── neg
│   │   │   └── pos
│   │   └── train
│   │       ├── all
│   │       ├── neg
│   │       ├── pos
│   │       └── unsup
│   ├── imdb_lm
│   └── tmp
├── wiki
│   ├── de-100
│   ├── de-100-unk
│   ├── de-2
│   ├── de-2-unk
│   ├── de-all
│   ├── _dumps
│   ├── _extr
│   │   └── de
│   │       ├── AA
│   │       ├── AB
...
│   │       └── CC
│   ├── wikitext-103
│   │   └── models
│   └── wikitext-2
└── xnli
    ├── __MACOSX
    │   ├── XNLI-1.0
    │   └── XNLI-MT-1.0
    │       ├── multinli
    │       └── xnli
    ├── XNLI-1.0
    └── XNLI-MT-1.0
        ├── multinli
        └── xnli
```

**Comment by PiotrCzapla** [comment]
@NirantK  I've merged your changes to `master`.  I also added few clean ups. You might want to regenerate your wiki text to follow the new structure:
```
data/wiki/de-100
├── de.wiki.test.tokens
├── de.wiki.train.tokens
└── de.wiki.valid.tokens
data/wiki/de-100-unk
├── de.wiki.test.tokens
├── de.wiki.train.tokens
└── de.wiki.valid.tokens
data/wiki/de-2
├── de.wiki.test.tokens
├── de.wiki.train.tokens
└── de.wiki.valid.tokens
data/wiki/de-2-unk
├── de.wiki.test.tokens
├── de.wiki.train.tokens
└── de.wiki.valid.tokens
data/wiki/de-all
├── de.wiki.test.tokens
├── de.wiki.train.tokens
└── de.wiki.valid.tokens
```
Or simply rmove files  ending with `.unk` and regenerate them as follows : 
```
find data -iname "*.unk"  # -delete once you see that it prints the right files
WIKI_DIR=data/wiki
LANG=??
python -m ulmfit.postprocess_wikitext "${WIKI_DIR}/${LANG}-2" $LANG
python -m ulmfit.postprocess_wikitext "${WIKI_DIR}/${LANG}-100" $LANG
```

**Comment by PiotrCzapla** [comment]
This partially fixes #2
